home/models.py

```diff
@@ -17,34 +19,36 @@ def __str__(self):
         return f"{self.name} | {self.email}"
 
     class Meta:
-        ordering = ('-created',)
-        constraints = [
-            models.UniqueConstraint(fields=['email', 'device_id'], name='Unique user + device')
-        ]
+        ordering = ("-created",)
+
 
 # Event model
 class DailyWalk(models.Model):
+    # Identifier
+    event_id = models.CharField(max_length=250, primary_key=True)
+    # Walk meta
     date = models.DateField()
     steps = models.IntegerField()
-    user = models.manufacturer = models.ForeignKey('AppUser',
-                                                   on_delete=models.CASCADE)
+    appuser = models.manufacturer = models.ForeignKey("AppUser", on_delete=models.CASCADE)
 
     def __str__(self):
-        return f"{self.user} | {self.date}"
+        return f"{self.appuser} | {self.date}"
 
     class Meta:
```

Comment:
Should we also add a compound uniqueness constraint between [appuser, date]...?

Reply:
I changed the primary key to the event ID you'd be sending so we should be good I presume. My guess is the event IDs you'd be sending are UUIDs? I am double checking to make sure the same date doesn't get overwritten unless there is an explicit update param passed

Reply:
Sure, I tend to always just add database-enforced validation whenever possible, as that also prevents errors in case writes are made that bypasses the stack. I.e. what if we end up writing some maintenance/helper scripts that do direct queries on the db, or someone needs to go in and manually fix some data... seems unlikely for our simple use case but you never know...

Reply:
Yeah that a good call! Would be a good fail safe when manually editing the database